Update dcnano lcdif display controller and rt500 board setting

KATE-WANG-NXP opened the pull request:
1. R500: Move the memory region definition to board dts. It is better to let the FLEXSPI2 memory region in the final linker file be generated from dts, in this way user/developer can place the data in the region in code more easily.
2. Update dcnano_lcdif driver. 
2.1 For the situation when the RAM space is limited and the driver's frame buffer cannot be place in RAM, add a new property 'ext-ram' in the binding, so if in device tree this property is assigned to a secondary RAM space, place the frame buffer there. The old way is to define the frame bufeer address in Kconfig, which may cause inconvenience if other data also needs to be placed in the same RAM space.
2.2 Update the driver to support new requirement on RT700. The IP requires a 64-byte alignment for the frame buffer stride.
2.3 Update the calculation of frame buffer size. If the pixel format is updated, the frame buffer size shall be updated too.
3. RT500/RT700:  Remove the external frame buffer address configuration for shield rk055hdmipi4ma0. In which memory region it shall be placed is obtained from device tree instead.
4. Enable the 720p panel RK055HDMIPI4MA0 for RT700 and add supported panels in board doc.

KATE-WANG-NXP committed:
boards: nxp: move memory region definition to board dts for RT500

It is better to let the FLEXSPI2 memory region in the final
linker file be generated from dts, in this way user/developer
can place the data in the region in code more easily.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
boards: shields: rk055hdmipi4ma0: Update RT500 configuration

Remove the external frame buffer configuration. It shall be set in
device tree instead now.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
boards: shields: rk055hdmipi4m: Update RT500 configuration

Remove the external frame buffer configuration. It shall be set in
device tree instead now.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

JarmouniA reviewed (commented):
You probably forgot to add `ext-ram = &psram;` to LCDIF DT node.

KATE-WANG-NXP commented:
Hi @JarmouniA , the frame buffer can be placed both on RAM and PSRAM, after test on PSRAM I found the demo cannot work properly, which can be reproduced on main, I reported an issue https://github.com/zephyrproject-rtos/zephyr/issues/100415
so in this PR I did not add the secondary ram configuration

JarmouniA reviewed (commented):
typo in the 1st commit title, `dedinition` -> `definition`.

KATE-WANG-NXP committed:
boards: nxp: Add memory region definition in board dts for RT700

Add memory region definition for PSRAM space which is controlled
by XPSI2 in board dts for RT700. In this way user/developer
can place the data in the region in code more easily. This space
can be used by media domain masters like DCNano lcdif controller.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
boards: shields: rk055hdmipi4ma0: Update RT700 configuration

Remove the external frame buffer configuration. It shall be set in
device tree instead now.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
samples: drivers: display: Enable new panel rk055hdmipi4ma0 for RT700

Add heap size configuration and add new panel rk055hdmipi4ma0 for RT700.
It requires extra space for 720p panel.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
drivers: display: dcnano_lcdif: Update framebuffer placement and pitch

1. For the situation when the RAM space is limited and the driver's
frame buffer cannot be place in RAM, add a new property 'ext-ram' in the
binding, so if in device tree this property is assigned to a secondary
RAM space, place the frame buffer there. The old way is to define the
frame bufeer address in Kconfig, which may cause inconvenience if other
data also needs to be placed in the same RAM space.
2. Update the driver to support new requirement on RT700. The IP requires
a 64-byte alignment for the frame buffer stride.
3. Update the calculation of frame buffer size. If the pixel format is
updated, the frame buffer size shall be updated too.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
doc: boards: nxp: Add new supported panels for RT700

Add new supported panels RK055HDMIPI4MA0 and ZC143AC72MIPI
in board doc.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
samples: drivers: display: Enable new panel zc143ac72mipi for RT700

Add zc143ac72mipi panel configuration for RT700 platform in display
sample test configuration.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>
